### ISSUE TITLE
[Bug]: `no-unregistered-classes` false-positive on Vue class binding

### ISSUE THREAD

**Issue by ShayanTheNerd** [schoero/eslint-plugin-better-tailwindcss#234]
```html
<script setup lang="ts">
const category = 'members' satisfies 'members' | 'vendors';
</script>

<template>
  1. <div :class="{ 'bg-primary': category === 'members' }"></div>
  2. <div :class="{ category === 'members' ? 'bg-primary' : undefined }"></div>
  3. <div :class="category === 'members' ? 'bg-primary' : undefined"></div>
  4. <div :class="[category === 'members' ? 'bg-primary' : undefined]"></div>
</template>
```

This rule works fine on `<div>`s 2, 3, and 4, but incorrectly reports `'members'` as an unregistered TailwindCSS class on the first `<div>`.

Here's my configuration object for eslint-plugin-better-tailwindcss:
```ts
import eslintPluginVue from 'eslint-plugin-vue';
import eslintPluginTailwind from 'eslint-plugin-better-tailwindcss';

const eslintParserVue = eslintPluginVue.configs['flat/recommended']
	.find((config) => config.name === 'vue/base/setup-for-vue')?.languageOptions?.parser;

const tailwindConfig = {
  name: 'tailwind',
  files: '**/*.vue',
  plugins: {
    'better-tailwindcss': eslintPluginTailwind,
  },
  languageOptions: {
    parserOptions: {
      parser: eslintParserVue,
    },
  },
  settings: {
    'better-tailwindcss': {
      entryPoint: './app/assets/styles/app.css',
      tsconfig: './tsconfig.json',

      /* For NuxtUI */
      attributes: [
        ['^v-bind:ui$', [
          { match: 'objectValues' },
        ]],
        ['^(?:v-bind:)?(class|activeClass|inactiveClass)$', [
          { match: 'strings' },
          { match: 'objectKeys' },
          { match: 'objectValues' },
        ]],
      ],
    },
  },
  rules: {
    /* Other rules */
    'better-tailwindcss/no-unregistered-classes': ['error', { detectComponentClasses: true }],
  },
};
```

**Comment by schoero** [comment]
I'm currently traveling. I will be able to look into it by the end of November.
